Dockerfile

```diff
@@ -13,8 +12,6 @@ RUN dpkg --add-architecture i386
 
 RUN if [ "$USE_TARGETID" = "ON" ] ; \
         then export ROCM_APT_VER=.apt_4.1.1;\
-    elif [ "$USE_MLIR" = "ON" ] ; \
```

Comment:
Maybe it's worth keeping capability of using different version of ROCm for MLIR builds. Are you sure that MLIR-enabled MIOpen will always use the same ROCm as non-MLIR one? If we upgrade our CI from 4.2 to 4.3, would MLIR work normally? If not, that would delay upgrading our CI until the MLIR issue is resolved. Of course we can simply switch off smoke MLIR testing by default in such a case... What do you think?

Reply:
Fair enough. I don't think I can give guarantee that MLIR implementation will always be forward and backward compatible. This PR itself is an example of how ROCm version bump can potentially break MLIR. So I agree with you that it might be easier to keep the branching logic here. I will make rocm version to be 4.2 to be consistent with the rest of the CI stages and leave it to be flexible in case future breakage comes in.

Reply:
Done in e8de8d3